### PR TITLE
Rdap query return multiple entity issue

### PR DESCRIPTION
**Pull request by maggarwal13**

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java [hunk 1]
```diff
@@ -117,41 +110,31 @@ public WhoisRdapService(final RdapQueryHandler rdapQueryHandler,
...
+                           @PathParam("objectType") RdapRequestType requestType,
                            @PathParam("key") final String key) {
 
         LOGGER.info("Request: {}", RestServiceHelper.getRequestURI(request));
+        Set<ObjectType> whoisObjectTypes = requestType.getWhoisObjectTypes(key);
 
-        switch (objectType.toLowerCase()) {
-            case "autnum": {
+        switch (requestType) {
+            case autnum: {
```
Comment (sbusk) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java]: By convention enum constants should be uppercased

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java [hunk 2]
```diff
@@ -117,41 +110,31 @@ public WhoisRdapService(final RdapQueryHandler rdapQueryHandler,
...
-                } else {
-                    whoisObjectTypes.add(PERSON);
-                    whoisObjectTypes.add(ROLE);
-                    whoisObjectTypes.add(MNTNER);
-                }
                 return lookupObject(request, whoisObjectTypes, key);
             }
-            case "nameserver": {
+            case nameserver: {
                 throw rdapExceptionMapper.notFound("nameserver not found");
```
Comment (sbusk) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java]: Why have all exceptions handled in the normal manner by the RdapExceptionMapper but for this case have the exceptionmapper build an exception which wraps the 404 and then throw this exception which is then caught by the exeptionhandler again as a WebApplicationException and just passed through?

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapExceptionMapper.java [hunk 3]
```diff
@@ -46,6 +47,18 @@ public Response toResponse(final Exception exception) {
             return Response.status(HttpServletResponse.SC_BAD_REQUEST).entity(createErrorEntity(HttpServletResponse.SC_BAD_REQUEST, exception.getMessage())).build();
         }
 
+        if (exception instanceof ParamException && ((ParamException) exception).getParameterName().equals("objectType")) {
+            return Response.status(HttpServletResponse.SC_BAD_REQUEST).entity(createErrorEntity(HttpServletResponse.SC_BAD_REQUEST, "unknown type")).build();
+        }
+
+        if (exception instanceof NotFoundException) {
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapExceptionMapper.java]: Are NotFound and BadRequest not already covered by the WebApplicationException catch-all immediately below?

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapRequestValidator.java [hunk 4]
```diff
@@ -8,59 +8,64 @@
...
     public void validateDomain(final String key) {
+        if (isEmpty(key)) {
+            throw new BadRequestException("empty lookup term");
+        }
+
         try {
             Domain.parse(key);
         } catch (AttributeParseException e) {
-            throw rdapExceptionMapper.notFound("RIPE NCC does not support forward domain queries.");
+            throw new NotFoundException("RIPE NCC does not support forward domain queries.");
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapRequestValidator.java]: were the badRequest and notFound methods redundant? I.e. is the response any different?

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java [hunk 5]
```diff
@@ -270,7 +239,8 @@ private Query getQueryObject(final Set<ObjectType> objectTypes, final String key
 
     private Response getResponse(HttpServletRequest request, List<RpslObject> result) {
         if (result.isEmpty()) {
-            throw rdapExceptionMapper.notFound("not found");
+            //throw rdapExceptionMapper.notFound("not found");
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java]: remove this redundant line

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapExceptionMapper.java (hunk 3 above)
Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapExceptionMapper.java]: As discussed, wrap Bad Request and Not Found in an RDAP JSON response. The WebApplicationException is generic, and should never happen, so log a warning and return the plain text error in that case.

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapRequestValidator.java (hunk 4 above)
Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapRequestValidator.java]: As discussed, these exceptions are handled in RdapExceptionMapper, and correctly wrapped in an RDAP-specific JSON response body.

File: whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java (hunk 2 above)
Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java]: This was resolved.

**Pull request merged**